infer/lib/python/inferlib/jwlib.py

```diff
@@ -24,14 +24,46 @@ def __init__(self, javac_args):
     def start(self):
         infer = os.path.join(config.BIN_DIRECTORY, 'infer')
         # pass --continue to prevent removing the results-dir
+        new_javac_args = []
+        list_count = 0
+        file_count = 0
+        source_list = "infer_sourcefiles"
+        if os.path.exists(source_list+str(list_count)):
+            os.remove(source_list+str(list_count))
+        f = open(source_list+str(list_count), "a")
+        for arg in self.javac_args:
+            if arg.endswith(".java"):
+                file_count += 1
+                f.write(arg + "\n")
+                if file_count >= 1000:
```

Comment:
If I understand correctly, this is putting the arguments that are "*.java" into files, with no more than 1000 of these per argument file.

I wonder if this is needed: could you just put all the arguments inside the same temporary file instead? That would create less churn on the filesystem and be much simpler.

Reply:
I will try it.